src/workerd/api/streams/readable.c++

```diff
@@ -108,12 +108,8 @@ jsg::Promise<ReadResult> ReaderImpl::read(
                   ") exceeds size of buffer (", options->byteLength, ").")));
         }
 
-        // TODO(conform): This should really be defined as the number of elements of the provided
-        // buffer view rather than a byte length for consistency with read().
-        // TODO(soon): This check is not valid for JS controllers, which support this no problem.
-        if (!options->bufferView.getHandle(js)->IsUint8Array()) {
-          KJ_LOG(WARNING, "Reading into non-Uint8Array isn't currently supported.");
-        }
+        jsg::BufferSource source(js, options->bufferView.getHandle(js));
+        options->atLeast = atLeast * source.getElementSize();
```

Comment:
This can *technically* be considered breaking and might be something we need a compat flag for. Specifically, before this change, if I called `reader.readAtLeast(2, new Uint16Array(2))`, the `atLeast` would be set to 2 bytes. After this change, it is set properly to 4 bytes. 

By making this change without a compat flag, I'm essentially hedging a bit on the warning there that, previously, "Reading into non-Uint8Array [wasn't] currently supported" because the number of bytes was not being correctly interpreted.